crypto/txscript/src/lib.rs

```diff
@@ -72,6 +72,66 @@ pub struct TxScriptEngine<'a, T: VerifiableTransaction> {
     num_ops: i32,
 }
 
+fn parse_script<T: VerifiableTransaction>(
+    script: &[u8],
+) -> impl Iterator<Item = Result<Box<dyn OpCodeImplementation<T>>, TxScriptError>> + '_ {
+    script.iter().batching(|it| {
+        // reads the opcode num item here and then match to opcode
+        it.next().map(|code| deserialize_opcode_data(*code, it))
+    })
+}
+
+pub fn get_sig_op_count<T: VerifiableTransaction>(signature_script: &[u8], prev_script_public_key: &ScriptPublicKey) -> u64 {
+    let is_p2sh = is_script_public_key_p2sh(prev_script_public_key.script());
+    let script_pub_key_ops = parse_script::<T>(prev_script_public_key.script()).collect_vec();
+    if !is_p2sh {
+        return get_sig_op_count_by_opcodes(&script_pub_key_ops);
+    }
+
+    let signature_script_ops = parse_script::<T>(signature_script).collect_vec();
+    if signature_script_ops.is_empty() || signature_script_ops.iter().any(|op| op.is_err() || !op.as_ref().unwrap().is_push_opcode()) {
+        return 0;
+    }
+
+    let p2sh_script = signature_script_ops.last().expect("checked if empty above").as_ref().expect("checked if err above").get_data();
+    let p2sh_ops = parse_script::<T>(p2sh_script).collect_vec();
+    get_sig_op_count_by_opcodes(&p2sh_ops)
+}
+
+fn get_sig_op_count_by_opcodes<T: VerifiableTransaction>(opcodes: &[Result<Box<dyn OpCodeImplementation<T>>, TxScriptError>]) -> u64 {
```

Comment:
I guess the golang "not precise" case is no longer needed.

Reply:
Right. It was also not used in golang